Source/SA/RFmx SpecAn/RFmxSpecAn.cs

```diff
@@ -44,33 +41,332 @@ public struct AutoLevelConfiguration
+                    FftPadding = 1,
+                    RbwAuto = RFmxSpecAnMXAcpRbwAutoBandwidth.True,
+                    Rbw_Hz = 10e3,
+                    RbwFilterType = RFmxSpecAnMXAcpRbwFilterType.Gaussian,
+                    SweepTimeAuto = RFmxSpecAnMXAcpSweepTimeAuto.True,
+                    SweepTimeInterval_s = 1e-3
+
+                    };
+            }
+        }
+        public struct AcpCarrierChannelConfiguration
+        {
+            public double IntegrationBandwidth_Hz;
+            public RFmxSpecAnMXAcpCarrierRrcFilterEnabled RrcFilterEnabled;
+            public double RrcAlpha;
+            public RFmxSpecAnMXAcpCarrierMode Mode;
+            public double Frequency;
+            public static AcpCarrierChannelConfiguration GetDefault()
+            {
+                return new AcpCarrierChannelConfiguration
+                {
+                    IntegrationBandwidth_Hz = 18e6,
+                    RrcFilterEnabled = RFmxSpecAnMXAcpCarrierRrcFilterEnabled.False,
+                    RrcAlpha = 0.22,
+                    Mode = RFmxSpecAnMXAcpCarrierMode.Active,
+                    Frequency = 0.00
+                };
+        
+            }
+        }
+        public struct AcpOffsetChannelConfiguration
+        {
+            public double IntegrationBandwidth_Hz;
+            public RFmxSpecAnMXAcpOffsetEnabled OffsetEnabled;
+            public double OffsetFrequency_Hz;
+            public RFmxSpecAnMXAcpOffsetSideband OffsetSideBand;
+            public RFmxSpecAnMXAcpOffsetPowerReferenceCarrier OffsetPowerReferenceCarrier;
+            public int OffsetPowerReferenceSpecificIndex;
+            public double RelativeAttenuation_dB;
+            public RFmxSpecAnMXAcpOffsetRrcFilterEnabled RrcFilterEnabled;
+            public double RrcFilterAlpha;
+
+            public static AcpOffsetChannelConfiguration GetDefault()
+            {
+                return new AcpOffsetChannelConfiguration
+                {
+                    IntegrationBandwidth_Hz = 18e6,
+                    OffsetEnabled = RFmxSpecAnMXAcpOffsetEnabled.True,
+                    OffsetFrequency_Hz = 20e6,
+                    OffsetSideBand = RFmxSpecAnMXAcpOffsetSideband.Both,
+                    OffsetPowerReferenceCarrier = RFmxSpecAnMXAcpOffsetPowerReferenceCarrier.Closest,
+                    OffsetPowerReferenceSpecificIndex = 0,
+                    RelativeAttenuation_dB = 0,
+                    RrcFilterEnabled = RFmxSpecAnMXAcpOffsetRrcFilterEnabled.False,
+                    RrcFilterAlpha = 0.22
+                };
+            }
+        }
+
+        public struct AmpmConfiguration
+        {
+            public double MeasurementInterval_s;
+            public double DutInputPower_dBm;
+            public ComplexWaveform<ComplexSingle> ReferenceWaveform;
+            public RFmxSpecAnMXAmpmReferenceWaveformIdleDurationPresent IdleDurationPresent;

```

Comment:
Could we replace both of these lines with a `SG.Waveform` type? This has the waveform data and the `IdleDurationPresent` property already embedded.
```suggestion
public ComplexWaveform<ComplexSingle> ReferenceWaveform;
            public SG.Waveform ReferenceWaveform.;
```

Reply:
We could do that but it would require to mix the SG types into the SA. Do we want to do that?

Reply:
Agreed, we would be mixing types. We have already set that precedent in the DPD module by inputting and outputting SG types. That may be a bit different because it already depends on the SG library for a different function.

Nonetheless, I think it provides better cohesion between the modules. Maybe the solution is to add it to the new common type library that we have been discussing? @smooresni is working on an addition that would add a _Common_ library that would live at the RDL namespace level that all modules could inherit from. This is probably a good one to migrate to that common library.

If you agree, we could for now just reference the definition from the SG module. We could open an issue to later update this module and others with the shared type.

Reply:
Since we'd like to encourage people to use the SG module to read the waveform from TDMS, I think it makes sense to use the SG module waveform type here instead.

Reply:
move into SG type